graphql_compiler/schema_transformation/split_query.py

```diff
@@ -34,6 +46,478 @@ def __init__(self, query_ast):
+
+    if new_type_coercion is not type_coercion:
+        return [new_type_coercion]
+    else:
+        return selections
+
+
+def _split_query_ast_one_level_recursive_normal_fields(
+    query_node, selections, type_info, edge_to_stitch_fields, name_assigner
+):
+    """One case of splitting query, selections contains a number of fields, no inline fragments.
+
+    The input selections will be divided into three sets: property fields, intra-schema vertex
+    fields, and cross-schema vertex fields.
+
+    Each cross-schema vertex field will not be included in the output selections. The AST
+    branch that each cross-schema vertex field leads to will be made into its own separate query
+    AST. The parent and child property fields used in the stich will be added to the parent and
+    child ASTs, if not already present. @outpt directives will be added to these parent and
+    child property fields, if not already present. @filter directives will not be added to
+    child property fields in this step. This is because one may choose to rearrange and reroot
+    the tree of SubQueryNodes to achieve an execution order with better performance. @filter
+    directives should be added only once the tree's structure is fixed.
+
+    _split_query_ast_one_level_recursive will be called recursive on each intra-schema vertex
+    field.
+
+    Args:
+        query_node: SubQueryNode, whose list of child query connections may be modified to
+                    include new children
+        selections: List[Field], containing a number of property fields and vertex fields
+        type_info: TypeInfo, used to get information about the types of fields while traversing
+                   the query AST
+        edge_to_stitch_fields: Dict[Tuple(str, str), Tuple(str, str)], mapping
+                               (type name, vertex field name) to
+                               (source field name, sink field name) used in the @stitch directive
+                               for each cross schema edge
+        name_assigner: IntermediateOutNameAssigner, object used to generate and keep track of
+                       names of newly created @output directives
+
+    Returns:
+        List[Field], with which to replace the list of selections in the SelectionSet one level
+        above. All cross schema edges in the input list will be removed, and in their place,
+        property fields added or modified. If no changes were made, the exact input list object
+        will be returned
+    """
+    parent_type_name = type_info.get_parent_type().name
+
+    property_fields_map, vertex_fields = _split_selections_property_and_vertex(selections)
+    intra_schema_fields, cross_schema_fields = _split_vertex_fields_intra_and_cross_schema(
+        vertex_fields, parent_type_name, edge_to_stitch_fields
+    )
+
+    made_changes = False
+    if len(cross_schema_fields) > 0:
+        made_changes = True
+
+    # First, no changes will be made to property fields
```

Comment:
I'd move this to right above line 184 and phrase it as `First, collect all property fields but don't make any changes to them yet` or something to that effect